graph/multi/weighted_directed_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright ©2014 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package multi
+
+import (
+	"testing"
+
+	"gonum.org/v1/gonum/graph"
+)
+
+var (
+	weightedDirectedGraph = (*WeightedDirectedGraph)(nil)
+
+	_ graph.Graph            = weightedDirectedGraph
+	_ graph.Directed         = weightedDirectedGraph
+	_ graph.WeightedDirected = weightedDirectedGraph
+)
+
+// Tests Issue #27
+func TestWeightedEdgeOvercounting(t *testing.T) {
+	g := generateDummyGraph()
```

Comment:
Should this use `generateDummyWeightedGraph`?

Reply:
Yes, thanks.

Done

Reply:
I think that `simple/weighted_directed_test.go` has the same oversight. You may want to fix it later.

Reply:
I'll fix that here and if you don't mind, squash the last two commits here into one so I can leave the three intended commits separate when I mergre. This will need another approval when it's done.